### DISCUSSION TITLE
DNS listens only on ipv6 despite listen_addresses is being set to ['0.0.0.0:53'] in config

### DISCUSSION THREAD

**Discussion by korn3r** [DNSCrypt/dnscrypt-proxy#2700] [Potential issues]
I [created issue](https://github.com/DNSCrypt/dnscrypt-proxy/issues/2699) about this. And issue has some "task" named "_Initially raised as discussion #..._"
...so, lets pretend im initially raising it as a discussion.
installed latest dnscrypt-proxy (prebuilt x86_64 binary) on Rocky 9.4
and dnscrypt proxy only listens on ipv6.
```
netstat -tulpn | grep :53
tcp6       0      0 :::53                   :::*                    LISTEN      2174/dnscrypt-proxy 
udp6       0      0 :::53                   :::*                                2174/dnscrypt-proxy 
```
```
cat dnscrypt-proxy.toml | grep ^listen_address
listen_addresses = ['0.0.0.0:53']
```

what does configuration file comments say about listen_addresses directive:
```
## To listen to all IPv4 addresses, use `listen_addresses = ['0.0.0.0:53']`
## To listen to all IPv4+IPv6 addresses, use `listen_addresses = ['[::]:53']`
```

**Comment by lifenjoiner** [comment]
Thanks for your feedback!
Already had been fixed by https://github.com/DNSCrypt/dnscrypt-proxy/commit/ac5087315c9c48eb1d113b42c69fa14e451a9a1c.

**Reply by korn3r** [reply]
Thanks. Any ETA on when new prebuilt binary will be available with this 7month old commit?
I know its probably not very good thing to rebuild everything very often, but 7 months is... a lot of time and this is pretty important fix.

**Reply by lifenjoiner** [reply]
You can build your own easily without waiting: https://github.com/DNSCrypt/dnscrypt-proxy/wiki/building-from-source#5-build-dnscrypt-proxy

**Reply by korn3r** [reply]
well... i tried to rebuild fedora src for rocky, and hit the facts that couple go libraries are missing in rocky repos. i tried to rebuild those. and after around 34 packages a faced the issue with dnscrypt requiring quic-go package, that is requirering some other package, which required couple other packages.... in the end there was package that is requiring google rpc go package, that is requiring previous package (the one was requiring rpc itself). funny loop requirements. thanks google i guess.
anyway i decided to not touch GO even with 15-foot stick and also i disabled ipv6 already anyway, so subject is not an issue for me now :)

and i would allow myself to recommend (feature suggestion maybe): rebuild at least once a year. prebuilt binaries are ~13 month old. they work, but when someone tells you issue you are facing was fixed 7 months ago in sources that sounds kinda weird. also i did check discussions\bugs for same bug, but i didnt even thought i should check for 7 months.

Thanks for nice support in any case.

**Reply by lifenjoiner** [reply]
> i tried to rebuild fedora src for rocky, and hit the facts that couple go libraries are missing in rocky repos

All third-party packages are integrated/vendored already, downloading should not be needed. Weird ...
Every time just a `go build -ldflags="-s -w"` works for me.

> ETA

I think @jedisct1 will release a new version in due time.
